### PR TITLE
Make IntoAttribute more flexible for Options

### PR DESCRIPTION
**Pull request by SleeplessOne1917**
This change should both remove repetition for implementing `IntoAttribute` for `Option`s and help futureproof things by making sure any option of an intoattribute will always also be intoattribute.

### CODE REVIEW

File: integrations/axum/src/lib.rs [hunk 1]
```diff
@@ -180,7 +180,7 @@ pub async fn generate_request_and_parts(
 ///
 /// This can then be set up at an appropriate route in your application:
 ///
-/// ```
+/// ```rust,ignore
```
Comment (SleeplessOne1917) [inline integrations/axum/src/lib.rs:183]: I needed to add these to make tests pass. Some tests seemingly unrelated to what I changed still seem to be failing.

**Comment by SleeplessOne1917** [comment]
I ran all the commands listed on the CONTRIBUTING page. Working on this has also inspired an issue for me to work on (#2131).

**Comment by gbj** [comment]
Could you clarify why `ignore`-ing unrelated tests is necessary here? If you rebase and revert those `ignore`s in Axum, do the tests pass?

**Comment by SleeplessOne1917** [comment]
@gbj I removed the ignores and now everything works 🤷

**Comment by gbj** [comment]
Okay, thanks! Looks good.

**Pull request merged**